mdata/aggregator.go

```diff
@@ -99,6 +100,51 @@ func (agg *Aggregator) flush() {
 	agg.agg.Reset()
 }
 
+// flush returns the aggregated point according to consolidator, and resets aggregation state
+func flush(aggregation *Aggregation, consolidator consolidation.Consolidator, currentBoundary uint32) schema.Point {
```

Comment:
I think that it would be a bit cleaner to separate the responsibilities a bit. I would add a member function called `GetValueFor` (or similar) like:

```
func (agg *Aggregator) GetValueFor(consolidator consolidation.Consolidator) float64 {
	switch consolidator {
	case consolidation.Cnt:
		return aggregation.Cnt
	case consolidation.Lst:
		return aggregation.Lst
	case consolidation.Min:
		return aggregation.Min
	case consolidation.Max:
		return aggregation.Max
	case consolidation.Sum:
		return aggregation.Sum
	}
        // panic? log error?
	return math.NaN()
}
```

And then in `Foresee` it would just look like:

```
aggregatedPoints = append(aggregatedPoints, schema.Point{Val:aggregationPreview.GetValueFor(consolidator), Ts: currentBoundary})
aggregationPreview.Reset()
```